### PR TITLE
Rebasing in changes #1340

### PR DESCRIPTION
**Pull request by Onkar627**
Documentation added related how to rebase in branches in contributing/git.

### CODE REVIEW

**Comment by Onkar627** [comment]
@programmer290399 is this fine?

**Comment by Onkar627** [comment]
I tried to add the image using this link Rebasing in branches but it appeared as link under the preview bar. Any suggestions to these changes @programmer290399 @pdxjohnny .

File: docs/contributing/git.rst [hunk 1]
```diff
@@ -104,6 +104,28 @@ At this point, all the CI tests should be passing. It should be okay for a
...
+
+1) Edit previous commit messages
+2) Combine multiple commits into one
+3) Delete or revert commits that are no longer necessary
+
+To rebase all the commits between another branch and the current branch state, you can enter the following command in your shell (either the command prompt for Windows, or the terminal for Mac and Linux):
+
+.. code-block:: console
+
+    $ git rebase --interactive other_branch_name
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:121]:
```suggestion
    $ git rebase --interactive other_branch_name
    
When you have an open PR and you want to take the changes that happened in main
and add them back to your PR, you can do the following.

.. image:: https://user-images.githubusercontent.com/74865775/159732897-c4ec72af-e456-4664-948a-5bc5397aa095.jpg
```

File: docs/contributing/git.rst [hunk 2]
```diff
@@ -104,6 +104,33 @@ At this point, all the CI tests should be passing. It should be okay for a
...
+    
+When you have an open PR and you want to take the changes that happened in main
+and add them back to your PR, you can do the following.
+
+.. image:: https://user-images.githubusercontent.com/74865775/159732897-c4ec72af-e456-4664-948a-5bc5397aa095.jpg
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:126]:
```suggestion
..
    sequenceDiagram
        upstream/main->>+feature_branch: git pull --rebase upstream main
        feature_branch->>+origin/feature_branch: git push --force
        origin/feature_branch->>+upstream/main: pull request

.. image:: https://user-images.githubusercontent.com/74865775/159732897-c4ec72af-e456-4664-948a-5bc5397aa095.jpg
```

File: docs/contributing/git.rst [hunk 3]
```diff
@@ -104,6 +104,39 @@ At this point, all the CI tests should be passing. It should be okay for a
 maintainer to merge your pull request at this point. Pull requests with the
 ``WIP:`` prefix or draft status will not be merged.
 
+Rebasing in branches
+--------------------
+
+The git rebase command allows you to easily change a series of commits, modifying the history of your repository. You can reorder, edit, or squash commits together.
+Git rebase is typically used to:-
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:111]:
```suggestion
The git rebase command allows you to easily change a series of commits,
modifying the history of your repository. You can reorder, edit, or squash
commits together. Git rebase is typically used to:
```

File: docs/contributing/git.rst [hunk 4]
```diff
@@ -104,6 +104,39 @@ At this point, all the CI tests should be passing. It should be okay for a
...
+
+1) Edit previous commit messages
+2) Combine multiple commits into one
+3) Delete or revert commits that are no longer necessary
+
+To rebase all the commits between another branch and the current branch state, you can enter the following command in your shell (either the command prompt for Windows, or the terminal for Mac and Linux):
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:117]:
```suggestion
To rebase all the commits between another branch and the current branch
state, you can enter the following command in your shell (either the
command prompt for Windows, or the terminal for Mac and Linux):
```

File: docs/contributing/git.rst [hunk 5]
```diff
@@ -104,6 +104,39 @@ At this point, all the CI tests should be passing. It should be okay for a
...
+
+..
+    sequenceDiagram
+        upstream/main->>+feature_branch: git pull --rebase upstream main
+        feature_branch->>+origin/feature_branch: git push --force
+        origin/feature_branch->>+upstream/main: pull request
+
+.. image:: https://user-images.githubusercontent.com/74865775/159732897-c4ec72af-e456-4664-948a-5bc5397aa095.jpg
+    
+To rebase the last few commits in your current branch, you can enter the following command in your shell:
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:134]:
```suggestion
To rebase the last few commits in your current branch, you can enter the
following command in your shell:
```

**Comment by Onkar627** [comment]
Referenced from - https://docs.github.com/en/get-started/using-git/about-git-rebase

File: docs/contributing/git.rst [hunk 6]
```diff
@@ -104,6 +104,39 @@ At this point, all the CI tests should be passing. It should be okay for a
 maintainer to merge your pull request at this point. Pull requests with the
 ``WIP:`` prefix or draft status will not be merged.
 
+Rebasing in branches
+--------------------
+
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:109]:
```suggestion
Sources/References:

- https://docs.github.com/en/get-started/using-git/about-git-rebase

```

**Comment by Onkar627** [comment]
@pdxjohnny is everything resolved for this PR or are there any changes required?

File: docs/contributing/git.rst [hunk 7]
```diff
@@ -104,6 +104,46 @@ At this point, all the CI tests should be passing. It should be okay for a
 maintainer to merge your pull request at this point. Pull requests with the
 ``WIP:`` prefix or draft status will not be merged.
 
+Rebasing in branches
+--------------------
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:108]:
```suggestion
--------------------

```

File: docs/contributing/git.rst [hunk 8]
```diff
@@ -104,6 +104,46 @@ At this point, all the CI tests should be passing. It should be okay for a
...
+Sources/References:
+
+- https://docs.github.com/en/get-started/using-git/about-git-rebase
```
Comment (pdxjohnny) [inline docs/contributing/git.rst:111]:
```suggestion
- https://docs.github.com/en/get-started/using-git/about-git-rebase

  - CC BY 4.0: Docs on this page were sourced from GitHub's open source
    docs. We've added some extra content intermixed, including a diagram
    showing the flow.
```

**Pull request merged**